### PR TITLE
Igx virtual directive

### LINKED ISSUE #482: Virtualization container
### Description  
Research and present initial design on a virtualization container.

#### The initial meeting resulted in the following sub-tasks to be worked on:

Research: 
- [x] Virtualization directive POC
- [ ] Virtualization limitations research
- [x] Virtualization component POC on top of the directive

### PR DESCRIPTION
**Pull request by MayaKirova**
Basic implemetation for igxVirtualDirective. Closes #482

### CODE REVIEW

**Comment by mpavlinov** [comment]
Closes #482

**Comment by ChronosSF** [comment]
@simeonoff , as discussed, the current flex rules for the grid doesn't allow for specific column widths that exceed the grid's own width (causing horizontal scrollbar) 

```css
.igx-grid__th--fw, .igx-grid__td--fw {
    -webkit-box-flex: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
}
```

Please, run the demos and check the Grid Performance sample.

**Comment by ChronosSF** [comment]
Before we merge to master we have to address the following:

- [ ]  Finish the spec for the virtualization directive
- [ ]  Tests
- [ ]  Polish for the igxList Performance sample in demos
- [ ]  Polish for the igxGrid Performance sample in demos
- [ ]  Fix the issue with the last column not getting rendered on the last position in certain scenarios based on the columns' widths
- [ ]  Attempt a no display-container approach to solve the no-virt scenario and the focus handling when navigating horizontally
- [ ]  Handle the grid width/height in % scenario.
- [ ]  Keyboard navigation to enter new virtualization frames
- [ ]  API fixes in virtual grids

**Comment by zdrawku** [comment]
@MayaKirova we've made a [pull request](https://github.com/IgniteUI/igniteui-angular/pull/561) related to demos update based on all components/directive naming changes and as you will notice the virtual-for is having a lot of warnings. I know that you are currently working on improvements and naming changes so we are going to leave these errors to be handled by your team once you are ready with the virtual-for implementation. 

Same applies for the conflicting files (caused by [this](https://github.com/IgniteUI/igniteui-angular/pull/541) pull request), although if you need any help with the conflicts please do let me know.

File: src/directives/virtual-for/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# igxVirtForOf
```
Comment (kdinev) [inline src/directives/virtual-for/README.md]: I guess the final naming decisions will affect the file names and the README.

**Pull request merged**